apps/btc_family/btc_txn.c

```diff
@@ -247,31 +286,175 @@ static bool fetch_transaction_meta(btc_query_t *query) {
       !check_which_request(query, BTC_SIGN_TXN_REQUEST_META_TAG)) {
     return false;
   }
-  // TODO: add checks for validating the metadata
+  // check important information for supported/compatibility
+  if (0x00000001 != query->sign_txn.meta.sighash ||
+      0 == query->sign_txn.meta.input_count ||
+      0 == query->sign_txn.meta.output_count) {
+    /** Do not accept transaction with empty input/output.
+     * Only accept SIGHASH_ALL for sighash type More info:
+     * https://wiki.bitcoinsv.io/index.php/SIGHASH_flags
+     */
+    btc_send_error(ERROR_COMMON_ERROR_CORRUPT_DATA_TAG,
+                   ERROR_DATA_FLOW_INVALID_DATA);
```

Comment:
Please document these error conditions in a document and put it up for review for the desktop team